### PR TITLE
Add `tvos` as a new platform

### PR DESCRIPTION
**Pull request by neonichu**

### CODE REVIEW

File: lib/cocoapods/installer/post_install_hooks_context.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ class UmbrellaTargetDescription
         #
         attr_accessor :specs
 
-        # @return [Symbol] The platform (either `:ios`, `:watchos` or `:osx`).
+        # @return [Symbol] The platform (either `:ios`, `:watchos`, `:tvos` or `:osx`).
```
Comment (segiddins) [inline lib/cocoapods/installer/post_install_hooks_context.rb]: Oxford comma.

**Comment by segiddins** [comment]
We probably need to override the code sign settings in the validator as well

File: lib/cocoapods/generator/xcconfig/xcconfig_helper.rb [hunk 2]
```diff
@@ -189,7 +189,7 @@ def self.add_developers_frameworks_if_needed(xcconfig, platform)
             search_paths = xcconfig.attributes['FRAMEWORK_SEARCH_PATHS'] ||= ''
             search_paths_to_add = []
             search_paths_to_add << '$(inherited)'
-            if platform == :ios || platform == :watchos
+            if platform == :ios || platform == :watchos || platform == :tvos
```
Comment (alloy) [inline lib/cocoapods/generator/xcconfig/xcconfig_helper.rb]: At some point one of these becomes prettier imo :)

```ruby
if [:ios, :watchos, :tvos].include?(platform)
  # ...
else
  # ...
end
```

or

```ruby
case platform
when :ios, :watchos, :tvos
  # ...
else
  # ...
end
```

**Comment by alloy** [comment]
Hot hot hot :+1:

File: lib/cocoapods/generator/header.rb [hunk 3]
```diff
@@ -84,6 +84,7 @@ def save_as(path)
       def generate_platform_import_header
         case platform.name
         when :ios then "#import <UIKit/UIKit.h>\n"
+        when :tvos then "#import <UIKit/UIKit.h>\n"
```
Comment (segiddins) [inline lib/cocoapods/generator/header.rb]: please no :(

Comment (neonichu) [inline lib/cocoapods/generator/header.rb]: I agree, but don't think this PR is the right place to change this stuff. Users will be very confused when UIKit symbols are suddenly undefined for just one of the UIKit platforms.

**Comment by segiddins** [comment]
:+1: but this needs a rebase

**Comment by segiddins** [comment]
I've rebased this, will merge if it passes

**Comment by orta** [comment]
:tada:

**Pull request merged**